### DISCUSSION TITLE
Kernel panic when booting from USB stick to install elementary OS 7

### DISCUSSION THREAD

**Discussion by peteruithoven** [elementary/triage#170] [Q&A]
I'm trying to upgrade my laptop to OS 7 but I'm running into a kernel panic. This while I can boot into Ubuntu 22.04 created in the same way, booting from the same USB stick. 

I'm on a Dell XPS 15 9550 (with NVIDIA GTX 960M). I'm using the same USB stick, USB port and Balena Etcher install.
I've used balenaEtcher-1.14.3-x64.AppImage
For both I went into bios to put usb drive as first option in the boot sequence. 

I've Downloaded Ubuntu 22.04.1 (ubuntu-22.04.1-desktop-amd64.iso) 
Selected: Try or install Ubuntu (first option)
Boots just fine into welcome install screen. 

Downloaded elementary OS (elementaryos-7.0-stable.20230129rc.iso)
(using magnet link)
I've verified the checksum following: https://elementary.io/docs/installation#verify-your-download
When booting I select the Try or install elementary OS option.

I get this error momentarily:
```
error: Out of memory.
Press any key to continue...
```

And then the following kernel panic:
```
[ 0.229634] x86/cpu: SGX disabled by BIOS,
[ 0.599320] ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PCI0.LPCB.H
EC.ECAV], AE_NOT_FOUND (20210730/psargs-330)
[ 0.599338] ACPI Error: Aborting method \_TZ.TZ00._TMP due to previous error
(AE_NOT_FOUND) (20210730/psparse-529)
[ 0.599588] ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PCI0.LPCB.H
EC.ECAV], RE_NOT_FOUND (20210730/psargs-330)
[ 0.599603] ACPI Error: Aborting method \_TZ.TZ00._TMP due to previous error
(AE_NOT_FOUND) (20210730/psparse-529)
[ 0.599705] ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PCI0.LPCB.H
EC.ECAV], AE_NOT_FOUND (20210730/psargs-330)
[ 0.599718] ACPI Error: Aborting method \_TZ.TZ01._TMP due to previous error
(AE_NOT_FOUND) (20210730/psparse-529)
[ 0.599812] ACPI BIOS Error (bug): Could not resolve symbol [\_SB.PCI0.LPCB.H
EC.ECAV], AE_NOT_FOUND (20210730/psargs-330)
[ 0.599824] ACPI Error: Aborting method \_TZ.TZ01._TMP due to previous error
(AE_NOT_FOUND) (20210730/psparse-529)
[ 0.697892] Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0)
[ 0.697968] CPU: 2 PID: 1 Comm: swapper/0 Not tainted 5.15.0-58-generic #64-Ubuntu
[ 0.698035] Hardware name: Dell Inc. XPS 15 9550/0N7TVV, BIOS 1.14.0 02/13/2020
[ 0.698099]  Call Trace:
[ 0.698122]  <TASK>
[ 0.698143]  show_stack+0x52/0x5c
[ 0.698181]  dump_stack_1v1+0x4a/0x63
[ 0.698217]  dump_stack+0x10/0x16
[ 0.698248]  panic+0x149/0x321
[ 0.698279]  mount_block_root+0x144/0x1dd
[ 0.698318]  mount_root+0x10c/0x11c
[ 0.698351]  prepare_namespace+0x13f/0x191
[ 0.698388]  kernel_init_freeable+0x18c/0x1b5
[ 0.698430]  ? rest_init+0x100/0x100
[ 0.698464]  kernel_init+0x1b/0x150
[ 0.698497]  ? rest_init+0x100/0x100
[ 0.698535]  ret_from_fork+0x1f/0x30
[ 0.698571]  </TASK>
[ 0.698641] Kernel Offset: 0x6800000 from 0xffffffff81000000 (relocation range: 0xffffffff80000000-0xffffffffbfffffff)
[ 0.698754] ---[ end Kernel panic - not syncing: VFS: Unable to mount root fs on unknown-block(0,0) ]---
```

I've tried reflashing and booting again. 
I've also tried the Save graphics option. 
Both fail. 

Ubuntu 22.04 is what elementary OS 7 is based on, I don't understand what could be going wrong here.

**Comment by peteruithoven** [comment]
I ran into https://github.com/elementary/installer/issues/542, which resulted in [this PR](https://github.com/elementary/os/pull/609). Building the ISO based on that PR following [these instructions](https://github.com/elementary/os#64-bit-amdintel), flashing that to an USB stick and then booting from it seems to work!

**Comment by philiplinden** [comment]
I am experiencing this problem on my XPS 13 9370 (4K Touchscreen). I attempted to apply the proposed fix to the grub config file (`gfxmode=800x600,auto`) and build from `main` on Ubuntu 22.04, but the problem persisted. I have not tried to build from https://github.com/elementary/os/pull/609 directly -- it is now pretty far behind.

In the meantime I installed Ubuntu 22.04 on this machine and have had no issues on that OS.
